src/tests/ftest/deployment/server_rank_failure.py

```diff
@@ -15,6 +16,20 @@
 from job_manager_utils import get_job_manager
 from run_utils import stop_processes
 
+HOST_GROUPS = []
+GROUP_1_HOSTS = list(NodeSet("aurora-daos-[0001-0128]"))
+GROUP_2_HOSTS = list(NodeSet("aurora-daos-[0129-0256]"))
+GROUP_3_HOSTS = list(NodeSet("aurora-daos-[0257-0384]"))
+GROUP_4_HOSTS = list(NodeSet("aurora-daos-[0385-0512]"))
+GROUP_5_HOSTS = list(NodeSet("aurora-daos-[0513-0640]"))
+GROUP_6_HOSTS = list(NodeSet("aurora-daos-[0641-0768]"))
+GROUP_7_HOSTS = list(NodeSet("aurora-daos-[0769-0896]"))
+GROUP_8_HOSTS = list(NodeSet("aurora-daos-[0897-1024]"))
+HOST_GROUPS = [
+    GROUP_1_HOSTS, GROUP_2_HOSTS, GROUP_3_HOSTS, GROUP_4_HOSTS, GROUP_5_HOSTS, GROUP_6_HOSTS,
+    GROUP_7_HOSTS, GROUP_8_HOSTS
+]
+
```

Comment:
This is the function we could use to avoid these hardcoded lists
```
def aurora_host_to_group(host):
    return ((int(re.findall(r'[0-9]+', host)[0]) - 1) // 128) + 1
```
And some examples
```
print(aurora_host_to_group('aurora-daos-0001'))
1
print(aurora_host_to_group('aurora-daos-0128'))
1
print(aurora_host_to_group('aurora-daos-0129'))
2
print(aurora_host_to_group('aurora-daos-1024'))
8
```

This function is really just a simple hash, so it could be swapped out for other functions on other systems, or for other requirements

Reply:
Done. Thanks!